llvm/include/llvm/CodeGen/TargetLowering.h

```diff
@@ -1675,8 +1675,8 @@ class TargetLoweringBase {
   /// operations except for the pointer size.  If AllowUnknown is true, this
   /// will return MVT::Other for types with no EVT counterpart (e.g. structs),
   /// otherwise it will assert.
-  EVT getValueType(const DataLayout &DL, Type *Ty,
-                   bool AllowUnknown = false) const {
+  virtual EVT getValueType(const DataLayout &DL, Type *Ty,
```

Comment:
I'd prefer to directly handle this in getValueType, rather than adding yet another target knob for really low level implementation details. Is this used in any meaningful way? Can we just return an MVT::Other or something in the unhandled case?

Reply:
I've added some handling to getValueType()

And ... I'm not sure. The getValueType() came in from the loop vectorizer wanting to know how many registers would be needed to hold some value, and that led to  the crash. It didn't help that, per comments from the last workaround of this sort I needed, there're a lot of pre-codegen uses of `MVT getPointerTYpe(unsigned AS);` that needed to be worked around with `MVT::v5i32`